### PR TITLE
[BEAM-2293] Rename Top.{Largest,Smallest=>Natural,Reversed}

### PR DESCRIPTION
**Pull request by wtanaka**
- [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by aaltay** [comment]
Hey @wtanaka PR description about exceptions and changes (to `Top`) looks unrelated. Could you update the PR description and title?

**Comment by wtanaka** [comment]
@aaltay done

**Comment by wtanaka** [comment]
rebased

**Comment by aaltay** [comment]
Thank you @wtanaka. Would not this rename be a breaking change in the stable API? Per the email thread responses, these could be introduced as new APIs instead (https://lists.apache.org/thread.html/9fb27e070b2038b5380dbff224a99c718790f6a3d42d51d9f84640eb@%3Cdev.beam.apache.org%3E).

**Comment by wtanaka** [comment]
agreed that since there was not much reply to Davor that it makes sense to do most conservative thing.

**Comment by aaltay** [comment]
Thank you @wtanaka. Once this PR is settled please consider making similar changes in Python SDK or filing a JIRA issue to do that.

R: @lukecwik as someone more familiar with Java API.

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Top.java [hunk 1]
```diff
@@ -327,14 +327,25 @@ private Top() {
   public static <K, V extends Comparable<V>>
       PerKey<K, V, List<V>>
       largestPerKey(int count) {
-    return Combine.perKey(new TopCombineFn<>(count, new Largest<V>()));
+    return Combine.perKey(new TopCombineFn<>(count, new Natural<V>()));
+  }
+
+  /**
+   * @deprecated use {@link Natural} instead
+   */
```
Comment (lukecwik) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Top.java]: Mark with `@Deprecated`

**Pull request closed without merging**